src/lib/assetBridger/erc20Bridger.ts

```diff
@@ -191,6 +219,76 @@ export class Erc20Bridger extends AssetBridger<
     return new Erc20Bridger(await getL2Network(l2Provider))
   }
 
+  /**
+   * Returns the addresses of all contracts that make up the gateway system
+   * @param gatewayAddress address of the gateway being queried (on L1 or L2)
+   * @param l1Provider An L1 provider
+   * @param l2Provider An L2 provider
+   * @returns Object containing the addresses of the contracts that make up the gateway system
+   */
+  public async getGatewayInformation(
+    gatewayAddress: string,
+    l1Provider: Provider,
+    l2Provider: Provider
+  ): Promise<ERC20GatewayInformation> {
+    // Checking providers
+    await this.checkL1Network(l1Provider)
+    await this.checkL2Network(l2Provider)
+
+    const gatewayInformation: ERC20GatewayInformation = {
+      l1GatewayRouter: '',
+      l2GatewayRouter: '',
+      l1ERC20Gateway: '',
+      l2ERC20Gateway: '',
+      l1Gateway: '',
+      l2Gateway: '',
+    }
+
+    try {
+      const l1Gateway = L1ArbitrumGateway__factory.connect(
+        gatewayAddress,
+        l1Provider
+      )
+
+      gatewayInformation.l1Gateway = gatewayAddress
+      gatewayInformation.l2Gateway = await l1Gateway.counterpartGateway()
+      gatewayInformation.l1GatewayRouter = await l1Gateway.router()
```

Comment:
`gateway.router()` is not part of the ITokenGateway interface, for example DAI and LPT gateway use `.l1router()`
in generally I don't think we need to return the router here, if needed they can be retrieved from the network object, you are also putting trust into the gateway to return the correct router address